src/accelerate/utils/modeling.py

```diff
@@ -648,7 +648,7 @@ def _get_named_modules(
             memo.add(module)
         yield prefix, module
         for name, sub_module in module._modules.items():
-            if module is None:
+            if sub_module is None:
```

Comment:
these function were copied from pytorch function so I don't think this is the right fix. Could you check if replacing `_get_named_parameters` in find_tied_parameters by `named_parameters` fixes the issue ? If not, it means that another fix is needed.

Reply:
I tried to use `named_parameters` instead of `_get_named_parameters`, but `named_parametes` doesn't have `remove_duplicate`, so it still fails at the next line.

Reply:
I tried the version `0.32.0`, and it works. To be honest, from the current code of `_get_named_modules`, module would definitively be None, as long as `module._modules` contains `None`. Would it be possible the problem lies in the model constructor itself? Usually, `module.modules` wouldn't contain None.

Reply:
> I tried to use named_parameters instead of _get_named_parameters, but named_parametes doesn't have remove_duplicate, so it still fails at the next line.

Reply:
After investigation the pytorch implementation, I confirm that we made a mistake when copy pasting and this is the right fix ! Thanks again for the fix !